src/core/dev_api/compare.hpp

```diff
@@ -0,0 +1,52 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+namespace ov {
+namespace cmp {
+/** \brief Enumerate bounds to compare */
+enum Bound : uint8_t { NONE, LOWER, UPPER, BOTH };
+
+/**
+ * \brief Compare if value is between lower and upper bounds.
+ *
+ * The Between comparator has four modes to check value:
+ * - Bound::None  (lower, upper)
+ * - Bound::LOWER [lower, upper)
+ * - Bound::UPPER (lower, upper]
+ * - Bound::BOTH  [lower, upper]
+ *
+ * \tparam T     Value type to compare.
+ * \tparam BMode Compare bounds mode.
+ */
+template <class T, Bound BMode = Bound::NONE>
+class Between {
```

Comment:
Why do we need this class in dev API? I think it can be a part of shape inference library.

Reply:
Moved.